### PR TITLE
Add Categories endpoint

### PR DESCRIPTION
**Pull request by 2e12**

### CODE REVIEW

File: backend/products/category.py [hunk 1]
```diff
@@ -22,6 +22,10 @@ def get_category_by_name(db, name):
     return db.query(Category).filter_by(name=name).first()
 
 
+def get_categories(db: Session, skip: int, limit: int):
+    return db.query(Category).order_by(Category.name.asc()).offset(skip).limit(limit).all()
```
Comment (LBruegger) [inline backend/products/category.py:26]: instead of limit(limit), I would suggest that you rename it to range

File: backend/products/routes.py [hunk 2]
```diff
@@ -18,6 +19,11 @@ def get_db():
     return db
 
 
+@product_router.get('/categories', response_model=List[product_schema.CategorySchema])
+async def get_all_categories(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
+    return get_categories(db, skip, limit)
```
Comment (LBruegger) [inline backend/products/routes.py:24]: Same here

**Review by LBruegger — CHANGES_REQUESTED** [review]
Overall it looks fine

**Comment by 2e12** [comment]
@LBruegger To keep it consistent, I would leave it as it is. The naming scheme has been used in other places as well.
Also the FastAPI documentation suggest this naming https://fastapi.tiangolo.com/tutorial/query-params/

**Comment by LBruegger** [comment]
> @LBruegger To keep it consistent, I would leave it as it is. The naming scheme has been used in other places as well.
> 
> Also the FastAPI documentation suggest this naming https://fastapi.tiangolo.com/tutorial/query-params/

I see. But Don't you agree that .limit(limit) is cumbersome?

**Comment by 2e12** [comment]
@LBruegger I have no problems with it. When we would agree to change, we have to refactor it in the complete project. Which would introduce breaking changes to the API. I'd prefer to stick to the convention.

**Comment by LBruegger** [comment]
> @LBruegger I have no problems with it. When we would agree to change, we have to refactor it in the complete project. Which would introduce breaking changes to the API. I'd prefer to stick to the convention.

No problem. Therefore leave it as it is, it's fine for me.

**Pull request merged**